### PR TITLE
Track log4j findings in Nessus and for customers

### PR DESCRIPTION
**Pull request by pburkholder**
## Changes proposed in this pull request:
-
- For each plugin related to log4j, count the expected ones, and report on the unexpected.
- From Nessus reports on diego cell paths, get to the responsible people on the customer side

## security considerations
This PR helps us track the findings that may still need attention, while dismissing the others.

### CODE REVIEW

**Comment by pburkholder** [comment]
I'm converting this to a draft. We should account for our false positives (mostly done) AND how we report on potential l4j findings to customers.  Patching customers now is far easier than support them in an IR situation.

**Comment by pburkholder** [comment]
Email sent:

```
<div class="aju"><div class="aCi"><img id=":lm_1-e" name=":lm" src="https://lh3.googleusercontent.com/a-/AOh14Gj6j0n2b9PO69Z7MLnKeWHfOLv2W_dD2EGGHqVy=s40" class="ajn" style="background-color: #cccccc" jid="peter.burkholder@gsa.gov" data-hovercard-id="peter.burkholder@gsa.gov" data-name="Peter Burkholder" aria-hidden="true"></div></div><div class="gs"><div class="gE iv gt">

Peter Burkholder <peter.burkholder@gsa.gov> | Peter Burkholder <peter.burkholder@gsa.gov> | Tue, Feb 8, 7:38 AM (1 day ago)
-- | -- | --
Peter Burkholder <peter.burkholder@gsa.gov>

Peter Burkholder <peter.burkholder@gsa.gov>
	
Tue, Feb 8, 7:38 AM (1 day ago)
	
to jgoetzin, mpadigal

Dear [cloud.gov](http://cloud.gov/) user:

We’re writing to you since you’re an Org Manager (see: https://landing.app.cloud.gov/docs/apps/managing-teammates/) for an application we’ve identified as potentially vulnerable to one or more of the recently identified log4j vulnerabilities -- see https://www.cisa.gov/uscert/apache-log4j-vulnerability-guidance for more information.

The table below lists the full filename, the affected application, space and org, as well as links to pages providing more information on the vulnerability. 

Maintaining the security of the code running on the [cloud.gov](http://cloud.gov/) Platform is a customer responsibility. Given the high-profile nature of these vulnerabilities, and the CISA ED 22_02 (https://www.cisa.gov/emergency-directive-22-02) we’re providing this notification in the interest of our customers.

We suggest updating your code and dependencies, or if the application is no longer needed, stopping or deleting the application. If you would like to delete the application and are unsure how to proceed, please forward this message to [support@cloud.gov](mailto:support@cloud.gov) and authorize us to delete the application on your behalf.

Thanks for your attention to this matter,

Peter Burkholder, [cloud.gov](http://cloud.gov/) Compliance and Security


App Name	File	Space Name	Org Name	Org Managers
```

**Comment by pburkholder** [comment]
I might need to use this for a few more months. The scripts could arguable go to internal-docs as part of a runbook. But it works for now.

And report-l4j-users really is a Python script that shells out to CFCLI and uses Perl. Am I proud of it? No. Could it be done wholly in Python? ¯\_(ツ)_/¯.  Probably not worth doing this time.

File: audit/log4j-nessus-parser.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# instance_guid and the app_guid.
+# 
+#   cat instance_guids | while read i; do
+#    if ( echo $i | grep -q -- - ); then
+#            jqcmd="jq '.LRPs[] | select (.instance_guid == \"$i\") | .process_guid'"
+#            echo -n "\"$i\", "
+#            cfdot cell-states | eval $jqcmd | cut -c1-36
+#    else
+#      echo "\"$i\", \"-\""
+#    fi
```
Comment (bengerman13) [inline audit/log4j-nessus-parser.py:24]: does this get run once total, on one diego cell? if so, could we get it somewhere else that doesn't require shelling out?

File: audit/log4j-nessus-parser.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                    if plugin_id not in path_info["plugins"]:
+                        path_info["plugins"].append(plugin_id)
+                    if report_host_name not in path_info["hosts"]:
+                        path_info["hosts"].append(report_host_name)
+                    path_report[path] = path_info
+
+
+# Path	Node 0	Instance GUID	Customer Path Name	Plugin Ids	App GUID	App Name	Space Name	Org Name	Org Manager	Space Developers																				
+
+import csv
```
Comment (bengerman13) [inline audit/log4j-nessus-parser.py:87]: if there's not a reason to defer this import, we should probably put it up at the top with the others

**Review by bengerman13 — APPROVED** [review]

**Comment by bengerman13** [comment]
one question and one nit to pick, but neither is a blocker and LGTM otherwise

File: audit/log4j-nessus-parser.py (hunk 1 above)
Comment (bengerman13) [reply audit/log4j-nessus-parser.py:24]: Prometheus and logsearch probably expose this (although that probably doesn't help directly).

Comment (pburkholder) [reply audit/log4j-nessus-parser.py:24]: I think we need to use `cfdot` to get that mapping. Does cfdot work on other nodes besides diego?

Comment (bengerman13) [reply audit/log4j-nessus-parser.py:24]: it seems plausible that we could configure it to run from a jumpbox, which seems like a slight improvement for the future

**Pull request merged**